core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -187,6 +188,9 @@ private DeciderOutcome decide(final Workflow workflow, List<Task> preScheduledTa
             if (!pendingTask.isExecuted() && !pendingTask.isRetried() && pendingTask.getStatus().isTerminal()) {
                 pendingTask.setExecuted(true);
                 List<Task> nextTasks = getNextTask(workflow, pendingTask);
+                if (pendingTask.isLoopOverTask() && !nextTasks.isEmpty()) {
```

Comment:
Here is the core logic. @apanicker-nflx While calling getnext task I am passing original ref name so taskmapper will work and while scheduling I change task ref name. So in general if task is part of loopOver Task then task will not be scheduled with original taskrefname which is different behaviour compare to existing.